### ISSUE TITLE
Number prefixed with dollar sign is replaced when using rangeToArray

### ISSUE THREAD

**Issue by sirbaconjr** [PHPOffice/PhpSpreadsheet#4242]
This is:

```
- [X] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
A cell that contains value like `200 number, $200 value` that is formatted as Text should be recognized as `200 number - $200 value` when using the `rangeToArray` function.

### What is the current behavior?
The cell is being recognized as `200 number - 0 value`.

### What are the steps to reproduce?

```php
<?php

require_once __DIR__ . '/../vendor/autoload.php';

$inputFileName = __DIR__ . '/../issue.dollar.sign.xlsx';
$spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($inputFileName);
$sheet = $spreadsheet->getActiveSheet();

print_r($sheet->toArray());
```

[issue.dollar.sign.xlsx](https://github.com/user-attachments/files/17905675/issue.dollar.sign.xlsx)

### What features do you think are causing the issue

- [X] Reader
- [ ] Writer
- [ ] Styles
- [ ] Data Validations
- [ ] Formula Calculations
- [ ] Charts
- [ ] AutoFilter
- [ ] Form Elements

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?

XLS isn't affected.
XLSX is affected.

### Which versions of PhpSpreadsheet and PHP are affected?

It seems all of them.

**Comment by sirbaconjr** [comment]
I'm working on a PR to fix the above issue.

I already replicated the issue on the tests, now just need to finish writing the fix for it.

Any tips would be welcome, even more if this issue was noticed before by someone else. I couldn't find any previous issues or pull requests that would relate to this.

**Comment by sirbaconjr** [comment]
I managed to isolate the issue up until this place:

https://github.com/PHPOffice/PhpSpreadsheet/blob/master/src/PhpSpreadsheet/Worksheet/Worksheet.php#L2832-L2840

If we don't format the data, the issue doesn't happen.

The replacement is happening on these lines:

https://github.com/PHPOffice/PhpSpreadsheet/blob/master/src/PhpSpreadsheet/Style/NumberFormat/Formatter.php#L128-L130

It seems that even though the column is text is being considered a number formatter and then something recognizes the dollar signs followed by numbers as a regex replacement variable

**Comment by sirbaconjr** [comment]
A solution is available on PR #4243
